clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -114,7 +114,7 @@ public Properties unusedProperties() {
     }
 
     public Map<String, Object> originals() {
-        Map<String, Object> copy = new HashMap<String, Object>();
+        Map<String, Object> copy = new RecordingMap<>();
```

Comment:
In unusedProperties(), we probably want to use the originals variable instead of the originals() method since we don't want to record those properties as used there.